lib/cookies.js

```diff
@@ -61,6 +61,22 @@ Cookies.prototype = {
       throw new Error('Cannot send secure cookie over unencrypted connection')
     }
 
+    if ( opts && opts.domain && Array.isArray(opts.domain) && req.headers.host ) {
```

Comment:
Checking `req.headers.host` will likely not work with anyone who is behind a proxy, which provides the user-facing host name in `X-Forwarded-Host`, any many other different headers, depending on configuration. We should provide some way to accommodate them.